.github/workflows/build-release.yml

```diff
@@ -1,13 +1,10 @@
-name: Publish release binaries
+name: Build binaries and publish release
 
-on:
-  push:
-    tags:
-      - 'v*'
+on: push
```

Comment:
I don't particularly want to run this on every PR or commit. Ideally it would run on a schedule (but no more than once per commit), and I could additionally trigger it manually. Of those two, however, I'd prefer to be able to trigger it manually.

Is it possible to trigger a workflow with an either/or condition? We could trigger on the branch pattern `release-*` and run without uploading, or trigger on the tag pattern `v*` and run with uploading.